MetaMask/metamask-extension
# Added fallback copy for when we're not able to retrieve a erc721 or erc1155 name in the setApprovalForAll screen

## Explanation

Currently, on `develop`, when we're **not able** to retrieve a name for the NFT in the setApprovalForAll screen, we use the `NFT` as the fallback copy. With this PR we added fallback copy for when we're **not able** to retrieve a erc721 or erc1155 name in the setApprovalForAll screen. Now fallback copy is `NFTs of this collection` where `this collection` is a link to that contract on the block explorer (currently, on `develop`, the `NFT` fallback is the link for it).

* Fixes #17516 

## Screenshots/Screencaps

### Before

#### When `giving` a setApprovalForAll allowance:

![215808104-8b497c82-2c46-4e03-ad4a-5a4a8279b46e](https://user-images.githubusercontent.com/92527393/223065354-c17bd0fc-7725-4321-a33b-809179842a20.png)

#### When `revoking` a setApprovalForAll allowance:

![215810395-2992341d-7929-4871-828f-5bdcbcd2012a](https://user-images.githubusercontent.com/92527393/223065401-a477abb5-6bca-44bd-b9f7-4b760fa1c911.png)

### After

#### When `giving` a setApprovalForAll allowance (when we're **able** to retrieve a erc721 or erc1155 name):

![Screenshot 2023-03-09 at 18 19 10](https://user-images.githubusercontent.com/92527393/224108039-79c70b2b-42eb-40fa-96b9-68d7d985f51d.png)

#### When `revoking` a setApprovalForAll allowance (when we're **able** to retrieve a erc721 or erc1155 name):

![Screenshot 2023-03-09 at 18 19 37](https://user-images.githubusercontent.com/92527393/224108475-a0c47178-c3d2-4189-bce4-1b460a3e8e0b.png)

#### When `giving` a setApprovalForAll allowance (when we're **not able** to retrieve a erc721 or erc1155 name):

![Screenshot 2023-03-09 at 18 17 44](https://user-images.githubusercontent.com/92527393/224108215-7590b9ef-7923-4516-8cc0-8ce53c7dbe35.png)

#### When `revoking` a setApprovalForAll allowance (when we're **not able** to retrieve a erc721 or erc1155 name):

![Screenshot 2023-03-09 at 18 18 15](https://user-images.githubusercontent.com/92527393/224108603-8a67e713-57ae-47de-9cc6-1e26e72b641f.png)

## Manual Testing Steps

1. Go to ZenAcademy (erc1155) https://etherscan.io/token/0xf64e6fb725f04042b5197e2529b84be4a925902c#writeContract
2. Click Write Contract
3. Connect MM
4. Click `SetApprovalForAll`
5. Set any operator (address) and for approved (bool) set `true` for `giving`  a setApprovalForAll allowance and `false` for `revoking` a setApprovalForAll allowance
6. Click on Write
7. See new fallback copy
